src/Integration/SonarAnalyzer/ActiveSolutionBoundTracker.cs

```diff
@@ -0,0 +1,76 @@
+//-----------------------------------------------------------------------
+// <copyright file="ActiveSolutionBoundTracker.cs" company="SonarSource SA and Microsoft Corporation">
+//   Copyright (c) SonarSource SA and Microsoft Corporation.  All rights reserved.
+//   Licensed under the MIT License. See License.txt in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using Microsoft.VisualStudio.Shell;
+using System;
+using System.ComponentModel.Composition;
+using SonarLint.VisualStudio.Integration.Persistence;
+
+namespace SonarLint.VisualStudio.Integration.SonarAnalyzer
+{
+    [Export(typeof(IActiveSolutionBoundTracker))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class ActiveSolutionBoundTracker : IActiveSolutionBoundTracker
+    {
+        private readonly IHost extensionHost;
+        private readonly IActiveSolutionTracker solutionTracker;
+
+        [ImportingConstructor]
+        public ActiveSolutionBoundTracker([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider)
+        {
+            this.extensionHost = serviceProvider.GetMefService<IHost>();
+            this.solutionTracker = serviceProvider.GetMefService<IActiveSolutionTracker>();
+
+            // todo check if both of these are needed or not:
+            this.extensionHost.VisualStateManager.BindingStateChanged += SolutionTracker_ActiveSolutionChanged;
```

Comment:
VisualStateManager does manage the visual state (we call SetBoundProject in order to update the UI). Please do not change the property.